embedded-graphics/src/image/image_bmp.rs

```diff
@@ -257,72 +261,82 @@ mod tests {
             for (y, row) in pattern.iter().enumerate() {
                 for (x, &expected_color) in row.iter().enumerate() {
                     let pos = Point::new(x as i32, y as i32);
-                    let pixel = iter.next().unwrap();
+                    let pixel = iter.next().ok_or(())?;
 
                     assert_eq!(pixel, Pixel(pos, expected_color));
                 }
             }
 
             assert!(iter.next().is_none());
-        };
+
+            Ok(())
+        }};
     }
 
     #[test]
-    fn colors_rgb555() {
-        test_pattern!(Rgb555, include_bytes!("../../tests/colors_rgb555.bmp"));
+    fn colors_rgb555() -> Result<(), ()> {
+        test_pattern!(Rgb555, include_bytes!("../../tests/colors_rgb555.bmp"))?;
+
+        Ok(())
     }
 
     #[test]
-    fn colors_rgb565() {
-        test_pattern!(Rgb565, include_bytes!("../../tests/colors_rgb565.bmp"));
+    fn colors_rgb565() -> Result<(), ()> {
+        test_pattern!(Rgb565, include_bytes!("../../tests/colors_rgb565.bmp"))?;
+
+        Ok(())
     }
 
     #[test]
-    fn colors_rgb888_24bit() {
+    fn colors_rgb888_24bit() -> Result<(), ()> {
         test_pattern!(
             Rgb888,
             include_bytes!("../../tests/colors_rgb888_24bit.bmp")
-        );
+        )?;
+
+        Ok(())
     }
 
     #[test]
     #[ignore]
-    fn colors_rgb888_32bit() {
+    fn colors_rgb888_32bit() -> Result<(), ()> {
         test_pattern!(
             Rgb888,
             include_bytes!("../../tests/colors_rgb888_32bit.bmp")
-        );
+        )?;
+
+        Ok(())
     }
 
     #[test]
-    fn colors_grey8() {
-        let image: ImageBmp<Gray8> =
-            ImageBmp::new(include_bytes!("../../tests/colors_grey8.bmp")).unwrap();
+    fn colors_grey8() -> Result<(), ()> {
+        let image: ImageBmp<Gray8> = ImageBmp::new(include_bytes!("../../tests/colors_grey8.bmp"))?;
 
         assert_eq!(image.size(), Size::new(3, 1));
 
         let mut iter = image.into_iter();
 
-        let p = iter.next().unwrap();
+        let p = iter.next().ok_or(())?;
```

Comment:
Mapping all errors to single error type doesn't make sense in a test IMO, just use `unwrap`.